app/models/expense.rb

```diff
@@ -0,0 +1,7 @@
+class Expense < ApplicationRecord
+  validates :name, presence: true
+  validates :amount, presence: true, numericality: { greater_than: 0 }
+
+  belongs_to :author, class_name: 'User'
+  belongs_to :category
```

Comment:
- [ ] Kindly make sure to correct the association between the `Expense` and `Group` to be a many-to-many relationship as shown in the ERD. 🙏 It's very important to practice writing many-to-many relationships using RoR because it'll get your hands better at creating correct migrations and references between the associated models as well as indexing and foreign keys. 😄

Reply:
Hi @MahmoudBakr23, 
Thanks for your review. To implement many to many relationship, I have to create another table. Or any other way. Could you please help me? 

Thanks again.

Reply:
Yes exactly, a join table that works between the two main ones. There's another way which is by using the has and belongs to many association. The difference between them is that the first one has its own model and table in case you need it for something more complex, but the second one doesn't, so what is the most applicable to use in your situation?

Thanks for mentioning me, and happy coding.

Reply:
@MahmoudBakr23 
Thanks for your valuable suggestion.

Reply:
well resolved! ✔️